storage/remote/otlptranslator/prometheus/metric_name_builder.go

```diff
@@ -322,3 +260,72 @@ func removeItem(slice []string, value string) []string {
 	}
 	return newSlice
 }
+
+// BuildMetricName builds a valid metric name but without following Prometheus naming conventions.
+// It doesn't do any character transformation, it only prefixes the metric name with the namespace, if any,
+// and adds metric type suffixes, e.g. "_total" for counters and unit suffixes.
+//
+// Differently from BuildCompliantMetricName, it doesn't pre-check the presence of unit and type suffixes.
+// If "addMetricSuffixes" is true, it will add them anyway.
+//
+// Please use BuildCompliantMetricName for a metric name that follows Prometheus naming conventions.
+func BuildMetricName(metric pmetric.Metric, namespace string, addMetricSuffixes bool) string {
+	metricName := metric.Name()
+
+	if namespace != "" {
+		metricName = namespace + "_" + metricName
+	}
+
+	if addMetricSuffixes {
+		mainUnitSuffix, perUnitSuffix := buildUnitSuffixes(metric.Unit())
```

Comment:
You can't use `buildUnitSuffixes` here, because it translates characters that are not alphanumerics or underscores.

Reply:
We can if we stop using `cleanUpUnit()` inside `buildUnitSuffixes()`, then we just need to call `cleanUpUnit()` inside `BuildCompliantMetricName()`

Reply:
Changing that could make it more complicated to verify that the code does the right thing. Maybe `buildUnitSuffixes` should instead take an optional callback function argument, for sanitizing the suffixes components. That way, the UTF-8 translation mode can avoid suffix sanitization.

Reply:
I took a different approach here, but I'm not strongly opinionated about it.

I've updated `buildUnitSuffixes` to only return the suffixes without any string manipulation; the callers are now responsible for Trimming, concatenating, and/or appending when necessary.

What do you think about this approach?

Reply:
I'm going to review soon, and consider whether there are any downsides.